src/hooks/useSubaccount.tsx

```diff
@@ -877,6 +878,49 @@ const useSubaccountContext = ({ localDydxWallet }: { localDydxWallet?: LocalWall
     [localDydxWallet, compositeClient]
   );
 
+  const getVaultAccountInfo = useCallback(async () => {
+    if (!compositeClient?.validatorClient) {
+      throw new Error('client not initialized');
+    }
+    const result = await compositeClient?.validatorClient.get.getMegavaultOwnerShares(dydxAddress);
+    if (result == null) {
+      return result;
+    }
+    return new DydxChainTransactions().parseToPrimitives(result);
```

Comment:
Up to your discrepancy but we can also use `abacusStateManager.chainTransactions.parseToPrimitives(result)` instead of instantiating a new class

Reply:
oh yeah, I feel like we should just make that a utility function

Reply:
ooooo I like that